### PR TITLE
podman: update to 3.1.1

### PR DESCRIPTION
**Pull request by oskarirauta**
Signed-off-by: Oskari Rauta <oskari.rauta@gmail.com>

Maintainer: Oskari Rauta / @oskarirauta
Compile tested: x86_64, server, recent SnapShot
Run tested: x86_64, server, recent snapshot, tested

Description:
Update podman to version 3.1.1

**Other changes**
 - Updated init script: stop pods and containers on shutdown.
 - Enable AppArmor support

### CODE REVIEW

**Comment by BKPepe** [comment]
Please document your changes in the commit description because this is not a simple bump as stated in the commit subject "update to version 3.1.1" as you are doing many other changes.

File: utils/podman/Makefile
Comment (BKPepe) [inline utils/podman/Makefile:57]: Why the previous issue was not okay for you?  While using `DEPENDS+=` there can be multiple dependencies.

Comment (oskarirauta) [reply utils/podman/Makefile:57]: Sorry. My bad. My "private" version wasn't updated yet to that change which I made to version that I updated here earlier. Committing soon..

Comment (oskarirauta) [reply utils/podman/Makefile]: @BKPepe fixed.

**Comment by oskarirauta** [comment]
> Please document your changes in the commit description because this is not a simple bump as stated in the commit subject "update to version 3.1.1" as you are doing many other changes.

Not that many.  Below is complete expanded list of changes and reasons to them.

I found out that If you change location of containers to persistent storage instead of tmpfs, starting them will fail unless they have been stopped. If this is the case that reboot has occurred before pods and containers have been stopped, they cannot be started, they have to be removed and re-created. Change in initscript tries to avoid that. Even if containers are running at tmpfs, this won't hurt. Still, if something happens and system hangs/reboots/etc, script won't save you from that. It's just a attempt to make things easier. Luckily, there's no real damage when containers "fail" as you are supposed to _mount_ dynamic content anyways..

I provided a pull request for AppArmor. Before provided python3-readline is required by AppArmor. If AppArmor is accepted, applications supporting it natively, should have it enabled- this also won't hurt or grow codebase too much either.

**Comment by BKPepe** [comment]
Now, it is better, thanks! The explanation could be added to the commit description as you already wrote it, but if you want to leave it as it is, I am fine with that as well.

However, we can not merge this first before apparmor is not merged.

**Review by BKPepe — APPROVED** [review]
Now, it is better, thanks! The explanation could be added to the commit description as you already wrote it, but if you want to leave it as it is, I am fine with that as well.

Otherwise it looks good to me, thanks! Once CI pass, I will merge it.

**Comment by oskarirauta** [comment]
> Now, it is better, thanks! The explanation could be added to the commit description as you already wrote it, but if you want to leave it as it is, I am fine with that as well.
> 
> Otherwise it looks good to me, thanks! Once CI pass, I will merge it.

If you insist.. But I think it's good that way, there's a good reasons why I made these minor changes and I am not sure at all if they require that much of explaining. Well, I'll change it for you..

**Comment by oskarirauta** [comment]
> > Now, it is better, thanks! The explanation could be added to the commit description as you already wrote it, but if you want to leave it as it is, I am fine with that as well.
> > Otherwise it looks good to me, thanks! Once CI pass, I will merge it.
> 
> If you insist.. But I think it's good that way, there's a good reasons why I made these minor changes and I am not sure at all if they require that much of explaining. Well, I'll change it for you..

Done

**Review by BKPepe — APPROVED** [review]

**Pull request merged**
